dbt/include/spark/macros/materializations/table.sql

```diff
@@ -41,7 +41,20 @@
 # --- Autogenerated dbt materialization code. --- #
 dbt = dbtObj(spark.table)
 df = model(dbt, spark)
-df.write.mode("overwrite").format("delta").saveAsTable("{{ target_relation }}")
+
+# make sure pandas exists
+import importlib.util
+import pyspark.pandas
+package_name = 'pandas'
+if importlib.util.find_spec(package_name):
+    import pandas
+    # convert to pyspark.sql.dataframe.DataFrame
+    if isinstance(df, pandas.core.frame.DataFrame):
+      df = spark.createDataFrame(df)
+    elif isinstance(df, pyspark.pandas.frame.DataFrame):
```

Comment:
I think this can happen **outside** the `if importlib.util.find_spec(package_name):`. Since it would run only for systems with pandas installed.

Reply:
Thanks @chamini2 -- I manually tested that suggestion and pushed a commit 👍